tests/domain/projections_test.py

```diff
@@ -6,18 +6,17 @@
 
 def test_current_settings():
     history = [
-        factories.Set(key="set-once", value="42"),
-        factories.Set(key="set-and-changed", value="42"),
-        factories.Changed(key="set-and-changed", new_value="43"),
-        factories.Set(key="set-and-unset", value="42"),
-        factories.Unset(key="set-and-unset"),
-        factories.Unset(key="never-set"),  # handles unexpected key
+        factories.Set(key="set-once", value="42", idx=0),
+        factories.Set(key="set-and-changed", value="42", idx=0),
+        factories.Changed(key="set-and-changed", new_value="43", idx=1),
+        factories.Set(key="set-and-unset", value="42", idx=0),
+        factories.Unset(key="set-and-unset", idx=1),
     ]
 
     settings = projections.current_settings(history)
 
     assert settings == {
-        "set-once": "42",
-        "set-and-changed": "43",
-        # "set-and-unset" is not set
+        "set-once": projections.Setting("42", idx=0),
+        "set-and-changed": projections.Setting("43", idx=1),
+        "set-and-unset": projections.Setting(None, idx=1),
```

Comment:
Question: Is it right for us to get a `Setting` here, where the setting is no longer set? Perhaps this wants a null-object?

Reply:
The None setting kind of _is_ the null object. I'm not sure how else to represent 'there is no setting with this name'. Especially when the code that 'set's the new value will need something with an index it can increment.